File: Laba2/8.ts
const nStr = prompt('Введите n: '); // Ввод
const n = parseInt(nStr || '0'); // Строка в число
if (n > 0 && n < 27) {
    for (let i = 1; i <= n; i++) {
        let current = 'A'.charCodeAt(0) + n - 1; // Буква
        let line = ''; 
        for (let j = 0; j < i; j++) {
            line += String.fromCharCode(current);
            if (j !== i - 1) {
                line += ' ';
            }
            current--;
        }
        console.log(line);
    }
} else {
    console.log('В английском алфавите всего 26 букв!');
}

File: Laba2/8.cpp
#include <iostream>
using namespace std;

int main() {
    int n;
    cout << "Введите n: ";
    cin >> n;
    if (n > 0 and n < 27){ // «Проверка на дурака»
        for (int i = 1; i <= n; i++) {
            char current = 'A' + n - 1; // Текущая буква
            for (int j = 0; j < i; j++) { // Количество повторений в строке
                cout << current;
                if (j != i - 1) {
                    cout << " ";
                }
                current--;
            }
            cout << endl;
        }
    } else cout << "В английском алфавите всего 26 букв!";
    return 0;
}
